core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -599,4 +639,81 @@ public void testUpdateSortOrder() {
     Assert.assertEquals("Should be nulls first",
         NullOrder.NULLS_FIRST, sortedByX.sortOrder().fields().get(0).nullOrder());
   }
+
+  @Test
+  public void testUpdateSchema() {
+    Schema schema = new Schema(0,
+        Types.NestedField.required(1, "y", Types.LongType.get(), "comment")
+    );
+    TableMetadata freshTable = TableMetadata.newTableMetadata(
+        schema, PartitionSpec.unpartitioned(), null, ImmutableMap.of());
+    Assert.assertEquals("Should use TableMetadata.INITIAL_SCHEMA_ID for current schema id",
+        TableMetadata.INITIAL_SCHEMA_ID, freshTable.currentSchemaId());
```

Comment:
Minor: The schema passed in uses the same ID as the one that is validated (`INITIAL_SCHEMA_ID=0`). I think to test this, it would be better to pass in a schema with a non-zero ID.

Reply:
Do you mean to change the schema id supplied to `TableMetadata.newTableMetadata` to be non-0? Actually in `TableMetadata` the input schema Id will be ignored and always assign from 0, so changing this will break the test... Although I think this current behavior should be fine since I think this method is only called when creating a new table, so there shouldn't be any existing schema within the table that will collide on schema ID.